src/components/confirm-leave-mixin.cjsx

```diff
@@ -0,0 +1,12 @@
+$ = require 'jquery'
+
+module.exports =
+  # getFlux: -> {store, actions}
+  # getId: -> id 
+  componentDidMount: ->
+    $(window).on("beforeunload", @confirmLeave)
```

Comment:
oh, and instead of jQuery you can use:

```coffee
window.addEventListener 'beforeunload', (e) ->
  msg = 'Are you sure?'
  (e or window.event).returnValue = msg     # Gecko and Trident
  msg                                        # Gecko and WebKit
```